storage/data-plane/src/main/java/com/microsoft/azure/storage/blob/RequestIDFactory.java

```diff
@@ -26,7 +26,7 @@ public RequestPolicy create(RequestPolicy next, RequestPolicyOptions options) {
         return new RequestIDPolicy(next, options);
     }
 
-    private final class RequestIDPolicy implements RequestPolicy {
+    private static final class RequestIDPolicy implements RequestPolicy {
```

Comment:
Why did this and other classes had to be made static?
How did this impact their usage?

Reply:
This addresses the spotbugs warning.
>  SIC: Should be a static inner class (SIC_INNER_SHOULD_BE_STATIC)
> This class is an inner class, but does not use its embedded reference to the object which created it.  This reference makes the instances of the class larger, and may keep the reference to the creator object alive longer than necessary.  If possible, the class should be made static. 

- The_ reason has been clarified in the above statement. 

- The private inner class here initialize new instance in the method of the outer class. To the current usage perspective, make the inner class static will not affect other classes as the new instance does not share with other classes (since we do not initial it like Outer.Inner() but new Inner()).   Other static changes also applied.

Reply:
okay, makes sense if they're inner classes.